tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -91,16 +91,18 @@ public RecordEntry Lookup(RecordEntry requestEntry, RecordMatcher matcher, IEnum
             // normalize request body with STJ using relaxed escaping to match behavior when Deserializing from session files
             RecordEntry.NormalizeJsonBody(requestEntry.Request);
 
+            RecordEntry entry = null;
+
             lock (Entries)
             {
-                RecordEntry entry = matcher.FindMatch(requestEntry, Entries);
+                entry = matcher.FindMatch(requestEntry, Entries);
                 if (remove)
                 {
                     Entries.Remove(entry);
                 }
```

Comment:
@kristapratico I'm going to reach out to you in the morning about what the test is doing. I doubt this code will actually do anything good for us unfortunately.